server/src/main/java/com/google/musicanalysis/cache/AnalysisCache.java

```diff
@@ -0,0 +1,86 @@
+package com.google.musicanalysis.cache;
+
+import com.google.musicanalysis.types.*;
+import java.lang.Object;
+import java.lang.ClassNotFoundException;
+import java.time.Instant;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.io.File;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.ObjectInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectOutputStream;
+import java.io.IOException;
+
+
+public class AnalysisCache {
+    private HashMap<String, AnalysisGroup> responseMap;
+    
+    public AnalysisCache(){
+        this.responseMap = new HashMap<String, AnalysisGroup>();
+    }
+
+    public void open() {
+        HashMap<String, AnalysisGroup> responseData = null;
+
+        int file = createFile(); // this will have use later.
+
+        try {
+            // .ser for serialized 
+            FileInputStream inFile = new FileInputStream("cachedData.txt");
+            ObjectInputStream inData = new ObjectInputStream(inFile);
+            responseMap = (HashMap<String, AnalysisGroup>) inData.readObject();
+            inData.close();
+            inFile.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return;
+        } catch (ClassNotFoundException c) {
+            c.printStackTrace();
+            return;
+        }
+    }
+
+    public void close() {
+        try {
+            FileOutputStream outFile = new FileOutputStream("cachedData.txt");
+            ObjectOutputStream outData = new ObjectOutputStream(outFile);
+            outData.writeObject(this.responseMap);
+            outData.close();
+            outFile.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void add(String requestUrl, AnalysisGroup responseData) {
+        this.responseMap.put(requestUrl, responseData);
+    }
+
+    public AnalysisGroup search(String requestUrl) {
+        // map.get() returns null if not match
+        return this.responseMap.get(requestUrl);
+    }
+
+    public void delete(String requestUrl) {
+        this.responseMap.remove(requestUrl);
+    }
+
+    private int createFile() {
+        try {   //logic here is going to improve which is why it's int
```

Comment:
Bad comment. Use a well structured TODO as mentioned in an earlier comment.

Also, I'm going to push you to use an enum here instead of punting that work. Using an enum should be trivial and drastically improves readability.

Reply:
I changed in a future pr here: https://github.com/googleinterns/step99-2020/pull/57/commits/b155dad2f2d1c297a04bd2e1a438eedc7c7c5a4d#diff-ed8ac0828b7698d78f6fdfab5002f25f

Do you want me to do it on this one too?

Reply:
I changed in a future pr here: https://github.com/googleinterns/step99-2020/pull/57/commits/b155dad2f2d1c297a04bd2e1a438eedc7c7c5a4d#diff-ed8ac0828b7698d78f6fdfab5002f25f

Do you want me to do it on this one too?

Reply:
Yes